File: Java/0921_Minimum_Add_to_Make_Parentheses_Valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int result = 0;
        Stack<Character> st = new Stack<Character>();
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if(c == '('){
                st.push(c);
            }
            else if(c == ')'){
                if(!st.isEmpty() && st.peek() == '('){
                    st.pop();
                }
                else{
                    st.push(c);
                }
            }
        }
        
        return st.size();
    }
}

File: CS/0921_Minimum_Add_to_Make_Parentheses_Valid.cs
public class Solution {
    public int MinAddToMakeValid(string s) {
        Stack<char> st = new Stack<char>();
        foreach(char c in s){
            if(c == '(' || st.Count == 0){
                st.Push(c);
            }
            else if(c == ')' && st.Peek() == '('){
                st.Pop();
            }
            else{
                st.Push(c);
            }
        }
        return st.Count;
    }
}
